src/main/java/com/cloudbees/jenkins/plugins/amazonecs/pipeline/ECSTaskTemplateStepExecution.java

```diff
@@ -174,16 +173,22 @@ private ECSTaskTemplateCallback(ECSTaskTemplate taskTemplate) {
          */
         protected void finished(StepContext context) throws Exception {
             Cloud c = Jenkins.get().getCloud(cloudName);
+            String parentLabel = taskTemplate.getInheritFrom();
             if (c == null) {
                 LOGGER.log(Level.WARNING, "Cloud {0} no longer exists, cannot delete task template {1}",
                         new Object[] { cloudName, taskTemplate.getTemplateName() });
                 return;
             }
             if (c instanceof ECSCloud) {
-                LOGGER.log(Level.INFO, "Removing task template {1} from cloud {0}",
-                        new Object[] { c.name, taskTemplate.getTemplateName() });
                 ECSCloud ecsCloud = (ECSCloud) c;
-                ecsCloud.removeDynamicTemplate(taskTemplate);
+                if (ecsCloud.isCustomTaskDefinition(parentLabel)){
+                    LOGGER.log(Level.INFO, "Don't remove custom task template from cloud");
```

Comment:
does it make sense to put the name / ARN of the task definition to the log?
Otherwise this log message might be very unclear, as people do not know what it relates to

Reply:
I think it's not bad to put the name / ARN of the task definition to the log.